arm/proofs/aes.ml

```diff
@@ -0,0 +1,393 @@
+
+let aes_mix_columns = new_definition `aes_mix_columns (op:(128)word) : (128)word =
+    let out00 = aes_mix_word op 0 8 16 24 in
+    let out10 = aes_mix_word op 8 16 24 0 in
+    let out20 = aes_mix_word op 16 24 0 8 in
+    let out30 = aes_mix_word op 24 0 8 16 in
+    let out01 = aes_mix_word op 32 40 48 56 in
+    let out11 = aes_mix_word op 40 48 56 32 in
+    let out21 = aes_mix_word op 48 56 32 40 in
+    let out31 = aes_mix_word op 56 32 40 48 in
+    let out02 = aes_mix_word op 64 72 80 88 in
+    let out12 = aes_mix_word op 72 80 88 64 in
+    let out22 = aes_mix_word op 80 88 64 72 in
+    let out32 = aes_mix_word op 88 64 72 80 in
+    let out03 = aes_mix_word op 96 104 112 120 in
+    let out13 = aes_mix_word op 104 112 120 96 in
+    let out23 = aes_mix_word op 112 120 96 104 in
+    let out33 = aes_mix_word op 120 96 104 112 in
+    (word_join:8 word->120 word->128 word) out33
+      ((word_join:8 word->112 word->120 word) out23
+        ((word_join:8 word->104 word->112 word) out13
+          ((word_join:8 word->96 word->104 word) out03
+            ((word_join:8 word->88 word->96 word) out32
+              ((word_join:8 word->80 word->88 word) out22
+                ((word_join:8 word->72 word->80 word) out12
+                  ((word_join:8 word->64 word->72 word) out02
+                    ((word_join:8 word->56 word->64 word) out31
+                      ((word_join:8 word->48 word->56 word) out21
+                        ((word_join:8 word->40 word->48 word) out11
+                          ((word_join:8 word->32 word->40 word) out01
+                            ((word_join:8 word->24 word->32 word) out30
+                              ((word_join:8 word->16 word->24 word) out20
+                                ((word_join:8 word->8 word->16 word) out10 out00)))))))))))))) `;;
+
+time (TEST [aes_mix_columns; aes_mix_word; FFmul02; FFmul03; input])
+`aes_mix_columns input`;;
+
+(* ========================================================================= *)
+(* AESE                                                                      *)
+(* ========================================================================= *)
+let aese = new_definition `aese (d:128 word) (n:128 word) =
+  aes_sub_bytes (aes_shift_rows (word_xor d n)) `;;
+
+TEST [aese; aes_shift_rows; aes_sub_bytes; aes_sub_bytes_select; joined_GF2; GF2]
+`aese (word 0xae6910a45715645a02502baaf5a826c9) (word 0xec882f3270973907d69635eea82d71)`;;
+
+(* ========================================================================= *)
+(* AESMC                                                                     *)
+(* ========================================================================= *)
+let aesmc = new_definition `aesmc (n: 128 word) = aes_mix_columns n`;;
+
+
+(************************************************)
+(** CONVERSIONS FOR AESE and AESMC             **)
+(************************************************)
```

Comment:
I usually leave thanks to the writer who wrote a prototype of a code, but... probably this is my style of communication. :)

Reply:
Added!